### PR TITLE
Register stable plugins in ActionModule

### PR DESCRIPTION
**Pull request by pgomulka**
Discovered (loaded from named_components.json) stable plugins are stored in StablePluginRegistry
In order for a plugin to be accessible it has to be registered in ActionModule. 
Stable plugins are implementing a new api (stable-plugin-api and plugin-api) and have to be wrapped with old api to allow the code to work together with the old plugins.
The plugins at the moment are expected to have only a no-arg constructor. 
relates https://github.com/elastic/elasticsearch/issues/88980

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/indices/analysis/AnalysisModule.java [hunk 1]
```diff
@@ -68,13 +70,18 @@ public final class AnalysisModule {
     private final HunspellService hunspellService;
     private final AnalysisRegistry analysisRegistry;
 
-    public AnalysisModule(Environment environment, List<AnalysisPlugin> plugins) throws IOException {
-        NamedRegistry<AnalysisProvider<CharFilterFactory>> charFilters = setupCharFilters(plugins);
+    public AnalysisModule(Environment environment, List<AnalysisPlugin> plugins, StablePluginsRegistry stablePluginRegistry)
+        throws IOException {
```
Comment (ChrisHegarty) [inline server/src/main/java/org/elasticsearch/indices/analysis/AnalysisModule.java:74]: Does it make sense to retain the 2-arg constructor, which can delegate to the 3-arg with a `new StablePluginsRegistry()`, as the third arg?

File: server/src/main/java/org/elasticsearch/common/NamedRegistry.java [hunk 2]
```diff
@@ -45,4 +45,10 @@ public <P> void extractAndRegister(List<P> plugins, Function<P, Map<String, T>>
             }
         }
     }
+
+    public void register(Map<String, T> collect) {
+        for (Map.Entry<String, T> entry : collect.entrySet()) {
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/common/NamedRegistry.java:50]: Could just be `collect.forEach(this::register)`?

File: server/src/main/java/org/elasticsearch/indices/analysis/wrappers/StableApiWrappers.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * A utility class containing methods that wraps the Stable plugin API with the old plugin api.
+ * Note that most old and stable api classes have the same names but differ in package name.
+ * Hence this class is avoiding imports and is using qualifying names
+ */
+public class StableApiWrappers {
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/indices/analysis/wrappers/StableApiWrappers.java:35]: Are we aiming for stable plugins to always be wrapped, or will the new API migrate to be a first-class API in ES?

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/indices/analysis/wrappers/StableApiWrappers.java:35]: Are we introducing top-level classes alongside the old plugin API to run stable plugins within ES, or is it all going through the existing `Plugin` API?

Comment (pgomulka) [reply server/src/main/java/org/elasticsearch/indices/analysis/wrappers/StableApiWrappers.java:35]: new api will be wrapped to allow to insert into the existing data structures in `AnalysisModule` so we can reuse the existing internals

File: server/src/main/java/org/elasticsearch/indices/analysis/AnalysisModule.java (hunk 1 above)
Comment (ChrisHegarty) [reply server/src/main/java/org/elasticsearch/indices/analysis/AnalysisModule.java:74]: na! just a q.

**Review by ChrisHegarty — APPROVED** [review]
LGTM

**Comment by thecoop** [comment]
LGTM

**Review by thecoop — APPROVED** [review]

**Pull request merged**